src/beta/resources/referral.js

```diff
@@ -0,0 +1,162 @@
+      cvc: creditCardCVC,
+    },
+  });
+
+  try {
+    const response = await request;
+
+    return response.body.id;
+  } catch (error) {
+    if (error.response && error.response.body) {
+      throw new RequestError(error.response.body, stripeCreditCardUrl);
+    }
+    throw error;
+  }
+}
+
+/**
+ * Send the Stripe credit card ID to EasyPost to add the card to the user's account.
+ * @param {API} api - The API client to use.
+ * @param {string} referralApiKey - The referral user's production API key.
+ * @param {string} stripeCreditCardId - Stripe credit card ID
+ * @param {string} priority - Whether to add the card as the 'primary' or 'secondary' card
+ * @returns {Object} - Response body (EasyPost payment method object)
+ */
+async function sendCardDetailsToEasyPost(api, referralApiKey, stripeCreditCardId, priority) {
+  const _api = getReferralApi(api, referralApiKey);
+  const params = { credit_card: { stripe_object_id: stripeCreditCardId, priority } };
+  const response = await _api.post('credit_cards', params);
+
+  return response.body;
+}
+
+export default (api) =>
+  class Referral extends base(api) {
+    static _name = 'Referral';
+
+    static _url = 'referral_customers';
+
+    static key = 'user';
+
+    static propTypes = propTypes;
+
+    /**
+     * Update the referral's email address.
+     * @param {string} referralUserId - The referral user's ID.
+     * @param {string} email - The new email address.
+     * @returns {Promise<boolean>} - Returns true if the referral was updated successfully, false otherwise.
+     */
+    static async updateEmail(referralUserId, email) {
+      try {
+        const newParams = { user: { email } };
+        await api.put(`${this._url}/${referralUserId}`, newParams);
+
+        return true;
+      } catch (error) {
+        return Promise.reject(error);
+      }
+    }
+
+    /**
+     * Add a credit card to the referral's account
+     * @param {string} referralApiKey - The referral user's production API key.
+     * @param {string} creditCardNumber - The credit card number.
+     * @param {string} creditCardExpirationMonth - The credit card expiration month.
+     * @param {string} creditCardExpirationYear - The credit card expiration year.
+     * @param {string} creditCardCVC - The credit card CVC.
+     * @param {string} primaryOrSecondary - Whether to add the card as 'primary' or 'secondary' payment method (defaults to 'primary')
+     * @returns {Promise<never>} - Response body (EasyPost payment method object)
+     */
+    static async addCreditCard(
+      referralApiKey,
+      creditCardNumber,
+      creditCardExpirationMonth,
+      creditCardExpirationYear,
+      creditCardCVC,
+      primaryOrSecondary = 'primary',
+    ) {
+      const stripeCreditCardId = await sendCardDetailsToStripe(
+        api,
+        creditCardNumber,
+        creditCardExpirationMonth,
+        creditCardExpirationYear,
+        creditCardCVC,
+      );
```

Comment:
Let's add try-catch here in the event the Stripe endpoint doesn't return anything, so we can throw an explicit error to the users.

Reply:
To Jack's point, we should be wrapping these individual functions here in try/catch and returning helpful errors to the user in the off chance something happens outside our control (Stripe is down/times out, the keys of the request/response change, etc)